lib/ash_admin/components/resource/show.ex

```diff
@@ -940,17 +940,18 @@ defmodule AshAdmin.Components.Resource.Show do
   end
 
   defp value!(value) do
-    data = Phoenix.HTML.Safe.to_iodata(value)
-
-    if is_binary(data) and !String.valid?(data) do
+    if is_binary(value) and !String.valid?(value) do
       "<binary data>"
     else
-      data
+      case Phoenix.HTML.Safe.impl_for(value) do
+        nil ->
+          Logger.debug("Failed to display value:\n#{inspect(value)}")
+          "<display error>"
+
+        _module ->
+          value
+      end
     end
-  rescue
```

Comment:
This isn't quite right because any encoding error in the protocol will crash the whole page. Instead, we should wrap the `to_iodata` result in '{:safe, result}` I believe.

Reply:
I mistakenly thought the rescue clause was only there to handle cases where the protocol wasn't implemented. I've updated the code as you suggested. However, I kept the binary data check before calling Safe.to_iodata, since I think it makes more sense to do it earlier. Wrapping with {:safe, data} as you suggested resolved the double escaping issue. Thanks!